java/client/src/main/java/glide/api/commands/StreamBaseCommands.java

```diff
@@ -52,4 +53,24 @@ public interface StreamBaseCommands {
      * }</pre>
      */
     CompletableFuture<String> xadd(String key, Map<String, String> values, StreamAddOptions options);
+
+    /**
+     * Trims the stream by evicting older entries.
+     *
+     * @see <a href="https://redis.io/commands/xtrim/">redis.io</a> for details.
+     * @param key The key of the stream.
+     * @param options Stream trim options.
+     * @return The number of entries deleted from the stream.
+     * @example
+     *     <pre>{@code
+     * // A nearly exact trimming of the stream to at least a length of 10, limited to 10 entries
+     * Long trimmed = client.xtrim("key", new StreamTrimOptions.MaxLen(false, 10L)).get();
+     * System.out.println("Number of trimmed entries from stream: " + trimmed);
+     *
+     * // An exact trimming of the stream by minimum id of "0-3", limit of 10 entries
+     * Long trimmed = client.xtrim("key", new StreamTrimOptions.MinId(true, "0-3", 10L)).get();
+     * System.out.println("Number of trimmed entries from stream: " + trimmed);
+     * }</pre>
+     */
+    CompletableFuture<Long> xtrim(String key, TrimLimit options);
```

Comment:
I think we should rename to `StreamTrimOptions`

Reply:
Do you prefer that?